Include/internal/pycore_tagged.h

```diff
@@ -0,0 +1,73 @@
+#ifndef Py_INTERNAL_TAGGED_H
+#define Py_INTERNAL_TAGGED_H
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#ifndef Py_BUILD_CORE
+#  error "this header requires Py_BUILD_CORE define"
+#endif
+
+#include <stddef.h>
+
+
+typedef union {
+    uintptr_t bits;
+} _Py_TaggedObject;
+
+#define Py_OBJECT_TAG (0b0)
+#define Py_OBJECT_TEST_TAG (0b1)
+
+#if defined(Py_OBJECT_TEST_TAG)
+    #define Py_CLEAR_TAG(tagged) ((PyObject *)(uintptr_t)((tagged).bits & (~Py_OBJECT_TEST_TAG)))
```

Comment:
Let's try to have symmetrical names for these sorts of operations, such as "tag" / "untag" or "pack" / "unpack", as opposed to "pack" / "clear".

"Clear" is particularly confusing because `Py_CLEAR` is similarly named, but a very different operation.